alf/networks/network.py

```diff
@@ -418,3 +418,28 @@ def wrap_as_network(net, input_tensor_spec):
         raise ValueError("input_tensor_spec is undefined for net of "
                          "type: %s" % type(net))
     return NetworkWrapper(net, input_tensor_spec)
+
+
+class BatchSquashNetwork(Network):
+    """Wrap a network so that it works on multiple batch dims.
+
+    Args:
+        network: the network to be wrapped
+        batch_dims: how many batch dims to squash before forward
+    """
+
+    def __init__(self,
+                 network: Network,
+                 batch_dims: int = 2,
+                 name: str = "BatchSquashNetwork"):
+        super().__init__(network.input_tensor_spec, network.state_spec, name)
+        assert isinstance(network, Network)
+        self._network = network
+        self._bs = alf.layers.BatchSquash(batch_dims)
+
+        self._output_spec = network.output_spec
```

Comment:
the output_spec is not same as the original output_spec

Reply:
> the output_spec is not same as the original output_spec

THis is intended to overwrite the automatically calculated output spec which includes all dims after the first dim. The reason is that BatchSquashNetwork is a special case that accepts multiple batch dims, and we want to keep this multiple dims for the output.